pkg/ingester/activeseries/active_series.go

```diff
@@ -404,6 +453,43 @@ func (s *seriesStripe) purge(keepUntil time.Time) {
 	}
 }
 
+// remove a single series from the stripe.
+// This is mostly the same logic from purge() but we decrement counters for a single entry instead of incrementing for each entry.
+// Note: we might remove the oldest series here, but the worst thing can happen is that we let run a useless purge() cycle later,
+// so this method doesn't update the oldestEntryTs.
+func (s *seriesStripe) remove(ref storage.SeriesRef) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	entry, ok := s.refs[ref]
+	if !ok {
+		// At the time of writing this, entry.deleted is always true when calling remove().
+		// If this changed, we should check that to avoid calling s.deleted.purge() too many times.
+		s.deleted.purge(ref)
+		// Was purged already.
+		// We can assume s.deleted.purge() was called already too.
+		return
```

Comment:
Is this logic correct? There's no ref in `s.refs` so why calling `s.deleted.purge(ref)`? Also the comments don't make much sense to me (the comments before `s.deleted.purge(ref)` don't match with the comments after it).

Reply:
This was nonsense. Fixed in https://github.com/grafana/mimir/pull/5678/commits/c22dc1b7845a999f433718729dfb2091c5461422 by removing the call and rewording the comment.